.github/workflows/django.yml

```diff
@@ -0,0 +1,70 @@
+name: "Iconica Standard Django CI Action"
+
+on:
+  workflow_call:
+    inputs:
+      python_version:
+        type: "string"
+        description: "Version of Python to use"
+        required: true
+      poetry_version:
+        type: "string"
+        description: "Version of Poetry to use when installing dependencies"
+        required: true
+      django_settings_module:
+        type: "string"
+        description: "The Django settings module to use"
+        required: true
+
+env:
+  SSH_PRV_KEY: "${{ secrets.SSH_PRV_KEY }}"
+
+jobs:
+  ruff_lint:
+    runs-on: "ubuntu-latest"
+    timeout-minutes: 5
+    steps:
+      - uses: "actions/checkout@v4"
+      - uses: "chartboost/ruff-action@v1"
+
+  test:
+    runs-on: "ubuntu-latest"
+    timeout-minutes: 5
+    steps:
+      - uses: "actions/checkout@v4"
+      - uses: "actions/setup-python@v5"
+        with:
+          python-version: "${{ inputs.python_version }}"
+      - uses: "shimataro/ssh-key-action@v2"
+        if: "${{ env.SSH_PRV_KEY != '' }}"
+        with:
+          key: "${{ secrets.SSH_PRV_KEY }}"
+          known_hosts: "${{ vars.SSH_KNOWN_HOSTS }}"
```

Comment:
the same holds for the vars and the secrets. Don't directly reference them but use the env:

Reply:
Secrets should _not_ be in env variables, that makes them readable by just reading the CI output. GitHub explicitly recommends _against_ it. Alright with the vars though.